extensions/control-plane/provision/provision-gcs/src/main/java/org/eclipse/edc/connector/provision/gcp/GcsProvisionExtension.java

```diff
@@ -42,29 +36,23 @@ public String name() {
         return "GCP storage provisioner";
     }
 
+    @Inject
+    private TypeManager typeManager;
+
+    @Inject
+    private Vault vault;
+
     @Override
     public void initialize(ServiceExtensionContext context) {
         var monitor = context.getMonitor();
 
-        var projectId = context.getConfig().getString(GCP_PROJECT_ID);
-        var storageClient = createDefaultStorageClient(projectId);
-        var storageService = new StorageServiceImpl(storageClient, monitor);
-        var iamService = IamServiceImpl.Builder.newInstance(monitor, projectId).build();
+        var gcpCredential = new GcpCredentials(vault, typeManager, monitor);
```

Comment:
Please document in a README how/when the different credentials will be used so that users can understand how they have to deploy the connector.